### PR TITLE
Add a package vignette for the JS events

### PR DESCRIPTION
**Pull request by yihui**

### CODE REVIEW

**Comment by jcheng5** [comment]
Can you be explicit about which events can be canceled, and which properties can be modified? Thanks.

File: vignettes/events.Rmd [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+A number of JavaScript events have been supported in Shiny since v0.13. These events can be used to keep track of the app progress, or even manipulate the values of inputs/outputs. All event names have the prefix `shiny:`, e.g., `shiny:connected`. We can listen to these events using `.on()` in jQuery, e.g.,
+
+```javascript
+$(document).on('shiny:connected', function(event) {
+  alert('Connected to the server');
+});
+```
+
+When an event is triggered in Shiny, the `event` object may have some additional properties that can be used to query or modify the information in Shiny, as we will see later in this document. Some events can cancel the process in Shiny, e.g., stop the propogation of an input or output change to the server. Such events include `shiny:inputchanged`, `shiny:message`, `shiny:value`, `shiny:error`, `shiny:updateinput`. To cancel the Shiny process, you can use `event.preventDefault()`, e.g.,
```
Comment (yihui) [inline vignettes/events.Rmd]: The events that can be canceled are listed here.

**Comment by yihui** [comment]
Okay, I'll include a summary section and list these events and properties.

**Comment by yihui** [comment]
@jcheng5 This PR is complete now.

**Pull request merged**